src/Nethermind/Nethermind.Core/ExecutionRequest/ExecutionRequest.cs

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using Nethermind.Core.Collections;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+
+namespace Nethermind.Core.ExecutionRequest;
+
+public enum ExecutionRequestType : byte
+{
+    Deposit = 0,
+    WithdrawalRequest = 1,
+    ConsolidationRequest = 2
+}
+
+public class ExecutionRequest
+{
+    public byte RequestType { get; set; }
+    public byte[]? RequestData { get; set; }
+
+    public override string ToString() => ToString(string.Empty);
+
+    public string ToString(string indentation) => @$"{indentation}{nameof(ExecutionRequest)}
+            {{{nameof(RequestType)}: {RequestType},
+            {nameof(RequestData)}: {RequestData!.ToHexString()}}}";
+}
+
+public static class ExecutionRequestExtensions
+{
+    public const int depositRequestsBytesSize = 48 + 32 + 8 + 96 + 8;
+    public const int withdrawalRequestsBytesSize = 20 + 48 + 8;
+    public const int consolidationRequestsBytesSize = 20 + 48 + 48;
+
+    public static int GetRequestsByteSize(this IEnumerable<ExecutionRequest> requests)
+    {
+        int size = 0;
+        foreach (ExecutionRequest request in requests)
+        {
+            size += request.RequestData!.Length + 1;
+        }
+        return size;
+    }
+
+    public static void FlatEncodeWithoutType(this IEnumerable<ExecutionRequest> requests, Span<byte> buffer)
+    {
+        int currentPosition = 0;
+
+        foreach (ExecutionRequest request in requests)
+        {
+            // Ensure the buffer has enough space to accommodate the new data
+            if (currentPosition + request.RequestData!.Length > buffer.Length)
+            {
+                throw new InvalidOperationException("Buffer is not large enough to hold all data of requests");
+            }
+
+            // Copy the RequestData to the buffer at the current position
+            request.RequestData.CopyTo(buffer.Slice(currentPosition, request.RequestData.Length));
+            currentPosition += request.RequestData.Length;
+        }
+    }
+
+    public static ArrayPoolList<byte[]> GetFlatEncodedRequests(
```

Comment:
Is this structure really needed in the block itself?

Reply:
do you mean, is there need to specify `ExecutionRequests` field in the block ? 

It is a workaround to to populate `ExecutionRequests` in `GetPayloadResultV4`, which is returned by `engine_getPaylaodV4`

Reply:
Yes, i can see that. I was thinking maybe we could isolate it to being in `IBlockProductionContext`  in the cache or something like that.

Reply:
Yep, I was thinking to isolate it, but it would've required to create a wrapper to a block so that result of https://github.com/NethermindEth/nethermind/blob/84e114c11ac7902b53681f37ce81c4c92dbf2111/src/Nethermind/Nethermind.Consensus/Producers/BlockProducerBase.cs#L205 contains the requests. Moreover,  changes at many places would have to be done, all the way up to the `blockProcessor` step. 

To avoid all of these complexities, I chose to create a new field under `Block` class